src/metabase/driver.clj

```diff
@@ -940,13 +940,26 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
-(defmulti truncate!
+(defmulti with-transaction*
```

Comment:
do you mean to keep both?

Reply:
also it feels like this could belong to `metabase.driver.sql-jdbc.execute` along with `do-with-connection-with-options`

Reply:
Not sure what you mean by both. Truncate has not been in a release yet, so OK to remove, and the new macro delegates to the  new function.

Since this method is part of the public API and used with the other methods in this namespace, it feels like it belongs here. I'll take a look at the other namespace though and mull it over, thanks for the tip.